load_balancer/status.js

```diff
@@ -0,0 +1,75 @@
+/* eslint import/no-dynamic-require: 0 */
+const { Logger } = require('../util/logger.js');
+const { check } = require('../util/environmentCheck.js');
+const request = require('request-promise-native');
+
+/* The environment variable LBCONFIG will contain the path to the config file.
+  The actual path for the config is "../deploy/configs/load_balancer/nodes_data_conf.json"
+  For the docker containers, the path is /etc/load_balancer/nodes_data_conf.json */
+check('LBCONFIG');
+
+const lbConfig = require(process.env.LBCONFIG);
+
+const loggerConfig = lbConfig.load_balancer;
+loggerConfig.cmd = 'log';
+
+const logger = new Logger(loggerConfig);
+
+async function getComponentStatus(node) {
+  const resultJson = Object.assign({}, node);
+  if (node.hostname !== undefined && node.port !== undefined && node.role !== undefined) {
+    try {
+      const body = await request.get(`https://${node.hostname}:${node.port}/connectionCheck`);
+      if (body.toString() === 'true') {
+        logger.debug(`Node at ${node.hostname}:${node.port} is up and running.`);
+        resultJson.status = 'up';
+      }
+    } catch (err) {
+      logger.error(`Error connecting to ${node.hostname}:${node.port}`);
+      logger.error(err);
+      resultJson.status = 'down';
+    }
+  }
+  return resultJson;
+}
+
+async function getPromises(nodes) {
```

Comment:
The previous name of this function sounds better. getPromises() name is too generic and unintuitive.